code/modules/mob/dead/new_player/new_player.dm

```diff
@@ -173,18 +173,18 @@
 		return 1
 
 	if(href_list["ready"])
-		var/tready = text2num(href_list["ready"])
-		//Avoid updating ready if we're after PREGAME (they should use latejoin instead)
-		//This is likely not an actual issue but I don't have time to prove that this
-		//no longer is required
-		if(SSticker.current_state <= GAME_STATE_PREGAME)
-			ready = tready
-		//if it's post initialisation and they're trying to observe we do the needful
-		if(!SSticker.current_state < GAME_STATE_PREGAME && tready == PLAYER_READY_TO_OBSERVE)
-			ready = tready
-			make_me_an_observer()
+		if((length_char(client.prefs.features["flavor_text"])) >= MIN_FLAVOR_LEN)
+			var/tready = text2num(href_list["ready"]) //Mandatory flavortext requirement for readyup/join - Yawet.
+			if(SSticker.current_state <= GAME_STATE_PREGAME)
+				ready = tready
+			//if it's post initialisation and they're trying to observe we do the needful
+			if(!SSticker.current_state < GAME_STATE_PREGAME && tready == PLAYER_READY_TO_OBSERVE)
+				ready = tready
+				make_me_an_observer()
+			return
+		else
+			to_chat(client.mob, span_danger("Your flavortext does not exceed our minimum of 130 characters."))
```

Comment:
This should be rewritten to be an early return.
eg: instead of
```
if(x)
    [...]
else
    [...]
```
do
```
if(!x)
    [...]
    return
if(...)
    [...]
```

Reply:
as for why, it helps keep the amount of indentations down, and generally helps with readability